### PR TITLE
Add initializer and basic loan validations

### PR DESCRIPTION
**Pull request by barrutko**

### CODE REVIEW

**Review by vanruch — APPROVED** [review]

**Review by MiksuJak — APPROVED** [review]

File: test/truefi2/TrueAssuranceFund.test.ts [hunk 1]
```diff
@@ -28,4 +39,13 @@ describe('TrueAssuranceFund', () => {
     await safu.liquidate(loan.address)
     expect(await token.balanceOf(safu.address)).to.equal(0)
   })
+
+  it('fails if loan is not defaulted', async () => {
+    await expect(safu.liquidate(loan.address)).to.be.revertedWith('TrueAssuranceFund: Loan is not defaulted')
+  })
+
+  it('fails if loan is not defaulted', async () => {
```
Comment (yuchenlintt) [inline test/truefi2/TrueAssuranceFund.test.ts:47]: Let's update this description (looks like you might've just copy/pasted the previous test description and forgotten to rename it).

**Review by yuchenlintt — APPROVED** [review]

File: contracts/truefi2/TrueAssuranceFund.sol [hunk 2]
```diff
@@ -5,14 +5,23 @@ import {SafeMath} from "@openzeppelin/contracts/math/SafeMath.sol";
 import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
 import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 
+import {Initializable} from "../common/Initializable.sol";
 import {ILoanToken2} from "./interface/ILoanToken2.sol";
 import {ITrueFiPool2} from "./interface/ITrueFiPool2.sol";
+import {ILoanFactory2} from "./interface/ILoanFactory2.sol";
 
-contract TrueAssuranceFund {
+contract TrueAssuranceFund is Initializable {
```
Comment (hal909) [inline contracts/truefi2/TrueAssuranceFund.sol:13]: Make UpgradeableOwnable

Comment (yuchenlintt) [reply contracts/truefi2/TrueAssuranceFund.sol:13]: To clarify, do you mean UpgradeableClaimable?

File: test/truefi2/TrueAssuranceFund.test.ts (hunk 1 above)
Comment (barrutko) [reply test/truefi2/TrueAssuranceFund.test.ts]: 😅

File: contracts/truefi2/TrueAssuranceFund.sol (hunk 2 above)
Comment (barrutko) [reply contracts/truefi2/TrueAssuranceFund.sol:13]: Would be nice to have a consensus on that 😁
@hal909

**Pull request merged**